### DISCUSSION TITLE
How to integrate authentication/authorization for elsa/home page?

### DISCUSSION THREAD

**Discussion by rairafaqatali** [elsa-workflows/elsa-core#727] [Q&A]
Hi Team,
I have integrated Elsa into my project. When I navigate on /elsa/home page, there are workflows UI for anonymous users.  Can I integrate authentication for a specific user on that page? Let's say I have three roles:
1- Admin,
2- Viewer
3- Editor
Now, I want to elsa/home page behave differently for all these three roles.

**Comment by sfmskywalker** [comment]
Hi! The Elsa Dashboard MVC area is no longer being developed on. The focus lies with the Elsa 2.0 Dashboard, which is created as a standalone Blazor application. 

It will have user & roles down the line, although no date has been planned for it yet. Until then, you might consider copying the Elsa 1 Dashboard MVC area into your current application, or if you want to migrate to Elsa 2.0, you could clone the Blazor dashboard app and add your own users & roles. 

Similarly, there will be an Orchard Core module that installs Elsa in Orchard Core. Orchard Core already has users, roles & permissions, so this might be a great potential solution too.

**Reply by rairafaqatali** [reply]
I have integrated Elsa dashboard 1.2.2.28. So, as /elsa/home is a public URL, so anyone can access this URL although I have implemented an authentication mechanism in my own app. So, how can I restrict users to access this page securely? I have cloned the MVC dashboard code, but the package Elsa.AutoMapper.Extensions are deprecated so I can't embed the code

**Reply by sfmskywalker** [reply]
That's odd. I don't see where Elsa 1.5 is currently using `Elsa.AutoMapper.Extensions`. Can you point me to it?

In any case, now that you have the dashboard source code integrated, you can use middleware to protect the Elsa routes from anonymous access, or even annotate the various controllers with `[Authorize]`.
